lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1180,18 +1197,16 @@ export class UserAgentApplication {
       }
       if (!this.redirectCallbacksSet) {
         // We reached this point too early - cache hash, return and process in handleRedirectCallbacks
-        self.cacheStorage.setItem(Constants.urlHash, hash);
+        this.cacheStorage.setItem(Constants.urlHash, locationHash);
         return;
       }
     }
 
-    self.processCallBack(hash, stateInfo, tokenResponseCallback);
+    this.processCallBack(locationHash, stateInfo, tokenResponseCallback);
 
     // If current window is opener, close all windows
-    if (isWindowOpenerMsal) {
-      for (let i = 0; i < window.opener.openedWindows.length; i++) {
-        window.opener.openedWindows[i].close();
-      }
+    if (isPopupOrIframe) {
+        window.openedWindows.forEach(openedWindow => openedWindow.close());
```

Comment:
Should we clean up iFrame here as well? `.close()` only close the popup window

Reply:
I'll take another look to make sure we are closing the iframes.